fgpyo/io/__init__.py

```diff
@@ -59,9 +59,19 @@
 from typing import Union
 from typing import cast
 
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
+
+
 COMPRESSED_FILE_EXTENSIONS: Set[str] = {".gz", ".bgz"}
 
 
+Reader: TypeAlias = Union[io.TextIOWrapper, TextIO, IO[Any]]
+Writer: TypeAlias = Union[IO[Any], io.TextIOWrapper]
```

Comment:
I added these type aliases to make it easier to reference the return types of `to_reader` and `to_writer` in other modules.

These were how the respective function signatures were defined, but is there any reason not to make them the same?

Reply:
Based on my understanding of the outputs, they should be `TextIO` or `io.TextIOWrapper`. The `IO[Any]` is superfluous.

Reply:
To be more specific, I don't think it's helpful to have overly broad types for _output_. It should be as specific as possible to help type-checking.